discretenet/problems/school_bus_scheduling/school_bus_scheduling.py

```diff
@@ -0,0 +1,185 @@
+                        for t_prime in range(t, min(t + m.R[s][r], all_time[-1] + 1))
+                    )
+                    for s, r in m.Rs
+                )
+                <= model.z
+            )
+
+        model.c5 = pyo.Constraint(model.T, rule=c5_rule)
+        self.model = model
+
+    def get_name(self):
+        return self.name
+
+    def get_parameters(self):
+        # to be implemented
+        return None
+
+
+class SchoolBusSchedulingGenerator(Generator[SchoolBusSchedulingProblem]):
+    def __init__(
+        self,
+        random_seed: int = 42,
+        path_prefix: Union[str, Path] = None,
+        num_routes=6,
+        max_time=120,
+        # school_start_interval=5,
+        num_schools=5,
+        time_window=20,
+        route_length_avg=30,
+        route_length_std=10,
+    ):
+        """
+        Initialize the school bus scheduling problem generator instance
+        following https://arxiv.org/abs/1803.09040v2
+        :param random_seed: The random seed to use
+        :param path_prefix: Path prefix to pass to instance ``save()`` methods
+            during batch generation. Must be set as a public instance attribute,
+            to be changed by the user after generator instantiation if desired.
+        :param num_routes: number of bus routes per school
+        :param max_time: total number of time slots ex: 120 for 2 hours
+        :param school_start_interval: school start time intervals
+            ex: 10 for 10, 20, 30 ... (Not implemented)
+        :param num_schools: number of schools
+        :param time_window: time window of acceptable bus arrival time
+        :param route_length_avg: average route length (time)
+        :param route_length_std: route length standard deviation
+        """
+        super().__init__(random_seed, path_prefix)
+        self.num_routes = num_routes
+        self.all_time = range(1, max_time)
+        # self.school_start_times = range(
+        #     school_start_interval,
+        #     max_time + school_start_interval,
+        #     school_start_interval,
+        # )
+        self.num_schools = num_schools
+        self.route_length_avg = route_length_avg
+        self.route_length_std = route_length_std
+        self.time_window = time_window
+        self.schools = None
+        self.routes = None
+        self.route_ids = None
+
+        self.name = "S{}_R{}_T{}_tw{}_ravg{}_rstd{}".format(
+            num_schools,
+            num_routes,
+            max_time,
+            time_window,
+            route_length_avg,
+            route_length_std,
+        )
+
+    def generate(self):
+        self.generate_schools()
+        self.generate_routes()
+        problem = SchoolBusSchedulingProblem(
+            self.all_time,
+            self.schools,
+            self.routes,
+            self.time_window,
+            self.name + "_%d" % self.random_seed,
+        )
+        return problem
+
+    def generate_schools(self):
```

Comment:
Have this return the list rather than setting `self.schools`.

Reply:
Updated